### DISCUSSION TITLE
[Feature Request] Specify token lifetime

### DISCUSSION THREAD

**Discussion by mbearup** [AzureAD/microsoft-authentication-library-for-python#775] [Q&A] [answered]
### MSAL client type

Confidential

### Problem Statement

As we work to harden our resources, one thing we've been investigating is token lifetime. We found that when we request a token using MSAL's [acquire_token_for_client](https://learn.microsoft.com/en-us/python/api/msal/msal.application.confidentialclientapplication?view=msal-py-latest#msal-application-confidentialclientapplication-acquire-token-for-client), we get back a token with a lifetime of one day. We'd really like to trim this back to 1-2 hours, but there seems to be no option to do so in MSAL/ConfidentialClientApplication.

Is there a plan to address this in the future, or is token lifetime managed some other way?

**Comment by rayluo** [comment, marked as answer]
Token lifetime is typically controlled by policy, not by how you call an sdk.
https://learn.microsoft.com/en-us/entra/identity-platform/configurable-token-lifetimes
